File: backend/Turistando/Turistando.WebApi/Models/Categoria.cs
using System.Collections.Generic;

namespace Turistando.WebApi.Models
{
    public class Categoria
    {
        public Categoria()
        {
            this.RoteiroCategorias = new List<RoteiroCategoria>();
        }

        public int Id { get; set; }
        public string Nome { get; set; }

        public virtual ICollection<RoteiroCategoria> RoteiroCategorias { get; set; }
    }
}

File: frontend/turistando-react/src/models/categoria.ts
import {RoteiroCategoria} from "./roteiro-categoria";

export class Categoria
{
    constructor()
    {
        this.Id = 0;
        this.Nome = '';
        this.RoteiroCategorias = new Array<RoteiroCategoria>();
    }

    public Id: number;
    public Nome: string;
    public RoteiroCategorias: RoteiroCategoria[];
}
